File: Documents/tareasPOO/Homework_1/hw_8/Usuario.py
import random

class Usuario:
    __id = 0
    __nombre = ""
    __edad = 0
    

    def __init__(self, nombre, edad):
        self.__id = random.randint(1, 1000)
        self.__nombre = nombre
        self.__edad = edad
        self.__libros_rentados = []

    def get_nombre(self):
        return self.__nombre
    
    def get_id(self):
        return self.__id
    
    def have_a_book(self):
        if len(self.__libros_rentados)!=0:
            return True
        else:
            return False
        
    def show_books(self):
        if len(self.__libros_rentados)!=0:
            print("Libros: ")
            for i in self.__libros_rentados:
                i.get_data()
    
    
    def rentar_libro(self,libro):
        self.__libros_rentados.append(libro)
        
    def devolver_libro(self,name):
        for  i in self.__libros_rentados:
          if(i.get_nombre()==name):
            h= self.__libros_rentados.index(i)
            self.__libros_rentados.pop(h)
    
    

File: Documents/tareasPOO/Homework_1/hw_8_py/hw8_jav/src/Usuario.java
import java.util.Random;
import java.util.ArrayList;

public class Usuario {
    Random ran = new Random();
    int __id = 0;
    String __nombre = "";
    int __edad = 0;
    ArrayList<Libro> __libros_rentados = new ArrayList<>();

    public Usuario(String nombre, int edad) {
        this.__id = ran.nextInt(1000);
        this.__nombre = nombre;
        this.__edad = edad;
    }

    public String get_nombre() {
        return this.__nombre;
    }

    public int get_id() {
        return this.__id;
    }

    public boolean have_a_book() {
        if (this.__libros_rentados.size() != 0) {
            return true;
        } else
            return false;
    }

    public void show_books() {
        if (this.__libros_rentados.size() != 0) {
            System.out.println("Libros: ");
            for (Libro libro : this.__libros_rentados) {
                libro.get_data();
            }
        }
    }

    public void rentar_libro(Libro libro) {
        this.__libros_rentados.add(libro);
    }

    public void devolver_libro(String name) {
        int h = -4;
        for (Libro libro : this.__libros_rentados) {
            if (libro.get_nombre().equals(name)) {
                h = this.__libros_rentados.indexOf(libro);
            }
        }
        if (h != -4) {
            if (h == 0) {
                this.__libros_rentados.clear();
            } else {
                this.__libros_rentados.remove(h);
            }

        }
    }
}
